src/Security/Authorization/Policy/src/AuthorizationMiddleware.cs

```diff
@@ -44,7 +60,24 @@ public AuthorizationMiddleware(RequestDelegate next, IAuthorizationPolicyProvide
     /// <param name="next">The next middleware in the application middleware pipeline.</param>
     /// <param name="policyProvider">The <see cref="IAuthorizationPolicyProvider"/>.</param>
     /// <param name="services">The <see cref="IServiceProvider"/>.</param>
-    public AuthorizationMiddleware(RequestDelegate next, IAuthorizationPolicyProvider policyProvider, IServiceProvider services) : this(next, policyProvider)
+    /// <param name="logger">The <see cref="ILogger"/>.</param>
```

Comment:
How do we feel about just service locating `ILogger<AuthorizationMiddleware>` so we don't have to add any new ctors, that would simplify the change a lot

Reply:
Only the longest constructor should be called by DI, I don't think we need the other new combinations. One new constructor isn't a big deal.

Reply:
@luccawilli can you remove all the extra new constuctor overloads except the one we need